core/decoders/zlib.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * noVNC: HTML5 VNC client
+ * Copyright (C) 2024 The noVNC Authors
+ * Licensed under MPL 2.0 (see LICENSE.txt)
+ *
+ * See README.md for usage and integration instructions.
+ *
+ */
+
+import Inflator from "../inflator.js";
+
+export default class ZlibDecoder {
+    constructor() {
+        this._zlib = new Inflator();
+        this._length = 0;
+    }
+
+    decodeRect(x, y, width, height, sock, display, depth) {
+        if ((width === 0) || (height === 0)) {
+            return true;
+        }
+
+        const pixelSize = depth == 8 ? 1 : 4;
+
+        if (sock.rQwait("ZLIB", 4)) {
+            return false;
+        }
```

Comment:
This should strictly speaking be inside the next if block, as you might have already read the length header.

Reply:
Thank you for the review. Fixed.